### ISSUE TITLE
Callback method when swipe animation is fully completed?

### ISSUE THREAD

**Issue by cliffmeyers** [h6ah4i/android-advancedrecyclerview#98]
We are trying to implement a feature that changes the UI for a row after its swipe has been completed (similar to how the UI changes after a message is dismissed in Gmail). I have been doing some testing with a few different callback methods:

RecyclerViewSwipeManager.OnItemSwipeEventListener.onItemSwipeFinished
SwipeableItemAdapter.onPerformAfterSwipeReaction

Unfortunately these seem to get called as soon as the user has swiped far enough to dismiss the item, not after the item has finished animating out of view. If I trigger the UI update immediately from within any of the above methods, I'm pretty confident the UI would look strange. Is there a different callback available to accomplish what I am looking for?

I did some additional investigation and found that ItemSliderAnimator uses a ViewPropertyAnimatorListener to do some cleanup in the onAnimationEnd method. This seems like a place where a callback method could be invoked to notify when a ViewHolder had completed its animation. It looks like it would not be difficult to wire this up between RecyclerViewSwipeManager and ItemSlidingAnimator.

I may be able to submit a PR for this work, provided I can clear it with my employer first.

**Comment by h6ah4i** [comment]
> I did some additional investigation and found that ItemSliderAnimator uses a ViewPropertyAnimatorListener to do some cleanup in the onAnimationEnd method. This seems like a place where a callback method could be invoked to notify when a ViewHolder had completed its animation. It looks like it would not be difficult to wire this up between RecyclerViewSwipeManager and ItemSlidingAnimator.

Yep, and also need to take care of Android 2.3 devices.
I would very appreciate if you could submit a PR :smile: Thanks.

**Comment by h6ah4i** [comment]
Hi. I have just published 0.8.0 and it has improved swipe event handling.

The [`SwipeResultAction.onSlideAnimationEnd()`](https://github.com/h6ah4i/android-advancedrecyclerview/blob/023f6c0cbcecc9f48f38622a7e8937b10b3d858c/library/src/main/java/com/h6ah4i/android/widget/advrecyclerview/swipeable/action/SwipeResultAction.java) method is called when swipe animation is fully complete. 

```java
class MySwipeResultAction extends 
        SwipeResultActionDefault
        (or SwipeResultActionMoveToSwipedDirection
        (or SwipeResultActionRemoveItem)
{

    /**
     * This method is called immediately after returning from the {@link SwipeableItemAdapter#onSwipeItem(RecyclerView.ViewHolder, int, int)} method.
     * You can modify data set and call notifyXXX() methods of adapter in this method.
     */
    @Override
    protected void onPerformAction() {
    }

    /**
     * This method is called when item slide animation has completed.
     */
    @Override
    protected void onSlideAnimationEnd() {
    }

    /**
     * This method is called after the {@link #onSlideAnimationEnd()} method. Clear fields to avoid memory leaks.
     */
    @Override
    protected void onCleanUp() {
    }
}
```

Please check the [new example implementation](https://github.com/h6ah4i/android-advancedrecyclerview/blob/f007b52c6e196e29c6142f7aa92791d6f520eb19/example/src/main/java/com/h6ah4i/android/example/advrecyclerview/demo_s/MySwipeableItemAdapter.java#L245) for more details.

Thanks.
